src/github/handlers/issue-comment-created.ts

```diff
@@ -42,6 +54,131 @@ const embeddedCommands: Array<OpenAiFunction> = [
   },
 ];
 
+async function buildPrompt(context: GitHubContext<"issue_comment.created">, commands: Array<OpenAiFunction>, manifests: Manifest[], similarExamples: string[]) {
+  // Gather command descriptions and examples
+  const availableCommands = commands.map((cmd) => ({
+    name: cmd.function.name,
+    description: cmd.function.description || "",
+    parameters: cmd.function.parameters,
+  }));
+
+  // Find matching examples and their command info from manifests
+  const detailedExamples = manifests.flatMap((manifest) => {
+    if (!manifest.commands) return [];
+
+    return Object.entries(manifest.commands).flatMap(([commandName, command]) => {
+      // Get all matching examples for this command
+      const allExamples = [];
+
+      // Add ubiquity:example first if it exists
+      if (command["ubiquity:example"]) {
+        allExamples.push({
+          commandInvocation: command["ubiquity:example"],
+          commandName,
+          description: command.description,
+          expectedToolCallResult: {
+            function: commandName,
+            parameters: command.parameters || {},
+          },
+        });
+      }
+
+      // Add any matching examples from the similar examples list
+      const matchingExamples = command.examples?.filter((example) => similarExamples.includes(example.commandInvocation)) ?? [];
+
+      allExamples.push(
+        ...matchingExamples.map((ex) => ({
+          ...ex,
+          commandName,
+          description: command.description,
+        }))
+      );
+
+      return allExamples;
+    });
+  });
+
+  // Format matched examples with their command info
+  const examplesSection = detailedExamples
+    .map((example) => {
+      const descriptionLine = example.description ? `Description: ${example.description}` : "";
+      return `Example: ${example.commandInvocation}
+Command: ${example.commandName}
+${descriptionLine}
+Tool Call:
+{
+  "function": "${example.expectedToolCallResult.function}",
+  "parameters": ${JSON.stringify(example.expectedToolCallResult.parameters, null, 2)}
+}`;
+    })
+    .join("\n\n");
+
+  // Format available commands section
+  const commandsSection = availableCommands
+    .map((cmd) => {
+      const commandDesc = cmd.description ? ` - ${cmd.description}` : "";
+      return `${cmd.name}${commandDesc}
+Parameters: ${JSON.stringify(cmd.parameters || {}, null, 2)}`;
+    })
+    .join("\n\n");
+
+  const systemMessage = `You are UbiquityOS, a GitHub bot that executes commands through function calls.
+
+Available Commands:
+${commandsSection}
+
+Similar Command Examples:
+${examplesSection}
+
+Input Format:
+{
+  "repositoryOwner": "string", // Repository owner's username
+  "repositoryName": "string",  // Repository name
+  "issueNumber": "number",     // Issue or PR number
+  "author": "string",         // Comment author's username
+  "comment": "string"         // The command text
+}
+
+Guidelines:
+1. Users invoke you with "@UbiquityOS" + command text
```

Comment:
Consider that we are about to run an experiment to map it to the issue author as well. 

There should be another task on this repo about it.